src/main.rs

```diff
@@ -1,3 +1,157 @@
+struct Mips {
+    regs: [u32; 32],
+    floats: [f32; 32],
+    mult_hi: u32,
+    mult_lo: u32,
+    pc: usize,
+
+    // Enough for six MIPS instructions.
+    memory: [u32; 13]
```

Comment:
Memory size will be constant across architectures, so 13 here should be some global constant

Reply:
yeah, this should be changed to a Vec for all memory regions.